### PR TITLE
Campaign dates should align

### LINKED ISSUE #145: Campaign bugs and issues
Campaign/Edit
Campaign/Create
Null name causes ongoing crashes in Admin Campaign tab
Managing Tenant → Managing Organization
Picture URL not accepting pictures
End date can be < Start date

### PR DESCRIPTION
**Pull request by MisterJames**
There are some complex validations that need to occur on the model (db lookup, multi-prop assertations). I wanted to use something like FluentValidation, but it's not ready for ASP.NET Core yet. This approach should do the trick.

 - Added validator
 - Moved validation from controller to validator
 - Fixed the validation for the start/end date

Closes #145

### CODE REVIEW

**Comment by MisterJames** [comment]
If someone gets a chance in the near future I'd love a review! (Here's looking at you @dpaquette or @mheggeseth :) )

File: AllReadyApp/Web-App/AllReady/Areas/Admin/Models/Validators/CampaignSummaryModelValidator.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            _mediator = mediator;
+        }
+
+        public async Task<Dictionary<string, string>> Validate (CampaignSummaryModel model)
+        {
+            var result = new Dictionary<string, string>();
+
+            if (model.EndDate < model.StartDate)
+            {
+                result.Add(nameof(model.EndDate), "The end date must fall after the start date.");
```
Comment (mheggeseth) [inline AllReadyApp/Web-App/AllReady/Areas/Admin/Models/Validators/CampaignSummaryModelValidator.cs]: It's actually _on or_ after the start date.

**Comment by mheggeseth** [comment]
@MisterJames Fix your validation message and you are good to go.

**Comment by MisterJames** [comment]
Great, thanks @mheggeseth! Fixed.

**Pull request merged**